tanner/tests/test_dorks_manager.py

```diff
@@ -0,0 +1,90 @@
+import unittest
+import asyncio
+import aioredis
+import random
+from unittest import mock
+from tanner import redis_client
+from tanner import config
+from tanner.utils.asyncmock import AsyncMock
+from tanner.dorks_manager import DorksManager
+
+
+class TestDorksManager(unittest.TestCase):
+    def setUp(self):
+        self.loop = asyncio.new_event_loop()
+        self.redis_client = None
+        self.path = None
+
+        async def connect():
+            self.redis_client = await redis_client.RedisClient.get_redis_client()
+
+        self.loop.run_until_complete(connect())
+
+        self.handler = DorksManager()
+        self.returned_result = None
+        self.expected_result = None
+
+    def test_push_init_dorks(self):
+
+        async def test():
+            await self.handler.push_init_dorks(config.TannerConfig.get('DATA', 'dorks'), self.handler.dorks_key,
+                                               self.redis_client)
+            self.returned_result = await self.redis_client.smembers(self.handler.dorks_key)
+
+        self.loop.run_until_complete(test())
+        self.assertEqual(len(self.returned_result), 3562)
+
+    def test_extract_path(self):
+        self.path = 'http://example.com/index.html?page=26'
+
+        async def test():
+            await self.handler.extract_path(self.path, self.redis_client)
+            self.returned_result = await self.redis_client.smembers(self.handler.user_dorks_key)
+
+        self.loop.run_until_complete(test())
+        self.expected_result = [b'http://example.com/index.html?page=']
+        self.assertEqual(self.returned_result, self.expected_result)
+
+    def test_extract_path_error(self):
+        self.path = '/index.html?page=26'
+        self.redis_client.sadd = AsyncMock(side_effect=aioredis.ProtocolError)
+
+        async def test():
+            await self.handler.extract_path(self.path, self.redis_client)
+
+        with self.assertLogs(level='ERROR') as log:
+            self.loop.run_until_complete(test())
+            self.assertIn('Problem with redis connection', log.output[0])
+
+    def test_init_dorks(self):
```

Comment:
Check the case if any key doesn't exist
https://github.com/mushorg/tanner/blob/master/tanner/dorks_manager.py#L48
https://github.com/mushorg/tanner/blob/master/tanner/dorks_manager.py#L49

Reply:
Done.

Reply:
No I meant to add a case when one or both have None values.

Reply:
Got it.